docs/product/roadmap.rst

```diff
@@ -1,6 +1,237 @@
+
+  - in an existing schema
+  - in a new schema
+- Create a table via importing data via
+   - CSV
+   - TSV
+   - copy/paste from a spreadsheet
+- View a table and see what type each column is.
+- Edit a table name
+- Edit a schema name
+- Edit the data in their table using a spreadsheet-like interface
+
+  - e.g. fill down cells by dragging a corner
+- Delete a table
+- Delete a schema
+
+At this point, all columns are displayed and edited as strings. We will support reflecting types from the database, but we're not going to be doing anything with those types yet.
+
+Filtering, Sorting, Grouping
+----------------------------
+Users should be able to perform these actions in both the GUI and API:
+
+- Filter the table by rules applied to data in various fields.
+- Sort the table by various fields.
+- Group results by the first character of any given field.
+- Group results by the first word of any given field.
+- Apply multiple filters, groups, and/or sorts at once.
+
+Multiple Types Support
+----------------------
+Users should be able to perform these actions in both the GUI and API:
+
+- Change the type of data per-column to any type supported by Postgres or PostGIS
+
+  - The data in the column will be validated.
+- Import data via any supported import method and have the types of data automatically detected during the import process
+
+  - Types are suggestions and can be changed by the user.
+
+Boolean Type
+------------
+- Add new type (use Postgres type if possible):
+
+  - Boolean
+- Autodetect this type during import
+- Allow user to change columns to this type
+- Add additional grouping options by value (yes/no)
+
+New Text Types
+--------------
+- Add new types:
+
+  - Email
+  - URL
+- Autodetect these types during import
+- Allow user to change columns to these types
+- Add additional grouping options:
+
+  - Email: Domain
+  - URL: TLD, Protocol
+
+Numeric Types
+-------------
+- Add new types, using existing Postgres types where possible:
+
+  - Money (with specified currency)
+  - Percentage
+  - Number
+- Autodetect these types during import
+- Allow user to change columns to these types
+- Add additional grouping options:
+
+  - Number: Range (calculate range options dynamically based on data, e.g. if data varies from 1-100, ranges could be 1-10, 10-20, etc.)
+
+  - Money: Range, Currency
+  - Percentage: Range
+
+Date and Time Types
+-------------------
+- Add new types, using existing Postgres types where possible:
+
+  - Date & Time
+  - Date
+  - Time
+  - Duration
+- Autodetect these types during import
+- Allow user to change columns to these types
+- Add additional grouping options:
+
+  - Date & Time, Date, Time support all grouping options supported by Postgres EXTRACT function.
+  - Duration: Range
+- Allow filtering using natural language for dates (e.g. "next month")
+
+Location Type
+-------------
+- Add new type, using existing PostGIS type where possible:
+
+  - Location
+- Autodetect this type during import
```

Comment:
I think we may want to see what we can get by default from PostGIS for this, and extend if necessary.  It may be that not _all_ possible location inferences will be reasonable in the MVP